docs/src/test/java/docs/http/javadsl/CustomHeaderExampleTest.java

```diff
@@ -0,0 +1,118 @@
+package docs.http.javadsl;
+
+import akka.http.javadsl.model.HttpHeader;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.ModeledCustomHeader;
+import akka.http.javadsl.model.headers.ModeledCustomHeaderCompanion;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.japi.JavaPartialFunction;
+import org.junit.Test;
+import scala.PartialFunction;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class CustomHeaderExampleTest extends JUnitRouteTest {
+  //#modeled-api-key-custom-header
+  public static class ApiTokenHeader extends ModeledCustomHeader {
+
+    ApiTokenHeader(String name, String value) {
+      super(name, value);
+    }
+
+    public boolean renderInResponses() {
+      return false;
```

Comment:
do we want to encourage defining custom "fake headers" that hold arbitrary request information that's not really a header? I always found it a bit of an anti-pattern for the existing cases where `akka-http` does it internally. Is there a good custom use case for it? If so, shouldn't they be sth like fully custom "request attributes" then instead?